### PR TITLE
Admin client API for writing forwarded metrics

### PR DESCRIPTION
**Pull request by xichen2020**
cc @cw9 @jeromefroe 

This PR adds the admin client interface as well as the API for writing forwarded metrics to destination aggregation servers.

There are other minor renaming changes due to consolidating metric types in `m3metrics` but most of them are immaterial.

### CODE REVIEW

File: aggregator/capture/aggregator.go [hunk 1]
```diff
@@ -87,6 +89,14 @@ func (agg *aggregator) AddUntimed(
 	return nil
 }
 
+// TODO(xichen): implement this.
+func (agg *aggregator) AddForwarded(
+	metric aggregated.Metric,
+	metadata metadata.ForwardMetadata,
+) error {
+	return nil
```
Comment (jeromefroe) [inline aggregator/capture/aggregator.go]: nit: would it be better to return a not implemented error just to be safe?

File: aggregator/list.go [hunk 2]
```diff
@@ -354,18 +354,18 @@ func (l *metricList) discardLocalMetric(
...
 }
 
-// consumeForwardMetric consumes a forward metric.
+// consumeForwardedMetric consumes a forward metric.
 // TODO(xichen): implement this.
-func (l *metricList) consumeForwardMetric(
+func (l *metricList) consumeForwardedMetric(
 	metric aggregated.Metric,
 	meta metadata.ForwardMetadata,
 ) {
```
Comment (jeromefroe) [inline aggregator/list.go]: nit: would it be worth panic'ing here just to ensure these functions aren't called until they are implemented?

File: aggregator/capture/aggregator.go (hunk 1 above)
Comment (xichen2020) [reply aggregator/capture/aggregator.go]: Sure can do, I'll add the implementation in the next PR but doesn't hurt either.

File: aggregator/list.go (hunk 2 above)
Comment (xichen2020) [reply aggregator/list.go]: Sure can do, I'll add the implementation in the next PR but doesn't hurt either.

**Pull request merged**